File: flutter_lottoresult/lib/data/repository/lotto_draw_result_repository.dart
import 'package:flutter_lottoresult/system/config.dart';
import 'package:flutter_lottoresult/data/model/lotto_draw_result.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:sqlite3/sqlite3.dart';

class LottoDrawResultRepository {
  final Database _db;
  final String _insertSql = '''
  INSERT INTO lotto_draw_result (
	drw_no,
	drw_no_date,
	drwt_no1,
	drwt_no2,
	drwt_no3,
	drwt_no4,
	drwt_no5,
	drwt_no6,
	bnus_no,
	tot_sell_amnt,
	first_win_amnt,
	first_prz_wner_co,
	first_accum_amnt,
	return_value
  ) values(?,?,?,?,?,?,?,?,?,?,?,?,?,?)
  ''';

  LottoDrawResultRepository(this._db);

  LottoDrawResult? getLottoDrawResultByDrwNo(int drwNo) {
    try {
      final ResultSet result = _db.select(
        "select * from lotto_draw_result where drw_no = ?",
        [drwNo],
      );
      return LottoDrawResult.fromDB(result.first);
    } catch (ex) {
      return null;
    }
  }

  void insertLottoDrawResult(LottoDrawResult result) {
    _db.execute(_insertSql, [
      result.drwNo,
      result.drwNoDate,
      result.drwtNo1,
      result.drwtNo2,
      result.drwtNo3,
      result.drwtNo4,
      result.drwtNo5,
      result.drwtNo6,
      result.bnusNo,
      result.totSellAmnt,
      result.firstWinAmnt,
      result.FirstPrzWnerCo,
      result.firstAccumAmnt,
      result.returnValue,
    ]);
  }

  void deleteLottoDrawResult(int drwNo) {
    _db.execute("delete from lotto_draw_result where drw_no = ?", [drwNo]);
  }
}

final sqlite3Provider = Provider<Database>((ref) {
  final config = ref.watch(configProvider);
  final database = sqlite3.open(config.dbFile);
  database.execute('''
create table if not exists lotto_draw_result (
        drw_no INT NOT NULL PRIMARY KEY,
        drw_no_date VARCHAR(200),
        drwt_no1 INT NOT NULL,
        drwt_no2 INT NOT NULL,
        drwt_no3 INT NOT NULL,
        drwt_no4 INT NOT NULL,
        drwt_no5 INT NOT NULL,
        drwt_no6 INT NOT NULL,
        bnus_no INT NOT NULL,
        tot_sell_amnt BIGINT,
        first_win_amnt INT,
        first_prz_wner_co INT,
        first_accum_amnt BIGINT,
        return_value VARCHAR(20)
)
''');
  return database;
});

final lottoDrawResultRepositoryProvider = Provider<LottoDrawResultRepository>((
  ref,
) {
  final sqlite3 = ref.watch(sqlite3Provider);
  return LottoDrawResultRepository(sqlite3);
});

// final lottoDrawResultRepositoryProvider = FutureProvider<LottoDrawResultRepository>((
//   
// ) async {
//   final sqlite3 = ref.watch(sqlite3Provider.future);
//   return LottoDrawResultRepository(sqlite3);
// });
//  

File: streamlit_lottoResult/app/data/repository/lotto_draw_result_repository.py
import sqlite3
from typing import Optional
from app.data.model import LottoDrawResult
from app.system import DB_NAME

class LottoDrawResultRepository:

    def _get_connection(self):
        conn = sqlite3.connect(DB_NAME)
        conn.row_factory = sqlite3.Row
        return conn

    def get_by_drw_no(self, drw_no: int) -> Optional[LottoDrawResult]:
        conn = self._get_connection()
        cursor = conn.cursor()
        cursor.execute("SELECT * FROM lotto_draw_result WHERE drwNo = ?", (drw_no,))
        row = cursor.fetchone()
        conn.close()
        if row:
            return LottoDrawResult(**row)
        return None

    def delete(self, drw_no: int) -> None:
        conn = self._get_connection()
        cursor = conn.cursor()
        cursor.execute("DELETE FROM lotto_draw_result WHERE drwNo = ?", (drw_no,))
        row = cursor.fetchone()
        conn.close()

    def create_table(self) -> None:
        conn = self._get_connection()
        cursor = conn.cursor()
        cursor.execute(
"""
create table if not exists lotto_draw_result (
        drwNo int not null primary key,
        drwNoDate varchar(200),
        drwtNo1 int not null,
        drwtNo2 int not null,
        drwtNo3 int not null,
        drwtNo4 int not null,
        drwtNo5 int not null,
        drwtNo6 int not null,
        bnusNo int not null,
        totSellamnt bigint,
        firstWinamnt int,
        firstPrzwnerCo int,
        firstAccumamnt bigint,
        returnValue varchar(20)
)
"""     )
        conn.commit()
        conn.close()

    def save(self, record: LottoDrawResult) -> None:
        conn = self._get_connection()
        cursor = conn.cursor()
        cursor.execute(
            """
            INSERT OR REPLACE INTO lotto_draw_result (
                drwNo,
                drwNoDate,
                drwtNo1,
                drwtNo2,
                drwtNo3,
                drwtNo4,
                drwtNo5,
                drwtNo6,
                bnusNo,
                totSellamnt,
                firstWinamnt,
                firstPrzwnerCo,
                firstAccumamnt,
                returnValue
            )
            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
            """,
            (
                record.drwNo,
                record.drwNoDate,
                record.drwtNo1,
                record.drwtNo2,
                record.drwtNo3,
                record.drwtNo4,
                record.drwtNo5,
                record.drwtNo6,
                record.bnusNo,
                record.totSellamnt,
                record.firstWinamnt,
                record.firstPrzwnerCo,
                record.firstAccumamnt,
                record.returnValue
            )
        )
        conn.commit()
        conn.close()
